pkg/api/handlers/mcp.go

```diff
@@ -829,17 +974,32 @@ func (m *MCPHandler) DeconfigureSharedServer(req api.Context) error {
 }
 
 func (m *MCPHandler) Reveal(req api.Context) error {
+	catalogID := req.PathValue("catalog_id")
+
 	var mcpServer v1.MCPServer
 	if err := req.Get(&mcpServer, req.PathValue("mcp_server_id")); err != nil {
 		return err
 	}
 
-	project, err := getProjectThread(req)
-	if err != nil {
-		return err
+	// For servers that are in catalogs, this checks to make sure that a catalogID was provided and that it matches.
+	// For servers that are not in catalogs, this checks to make sure that no catalogID was provided. (Both are empty strings.)
+	if mcpServer.Spec.SharedWithinMCPCatalogName != catalogID {
+		return types.NewErrNotFound("MCP server not found")
+	}
```

Comment:
This is a perfect example of why I don't like reusing API route handler functions across multiple routes like I am doing in this PR.

Without this check, there is a danger that the user could abuse the `/assistants/{assistant_id}/projects/.../{mcp_server_id}/reveal` route to access the credential belonging to a server that is actually in a catalog that they have access to, when only the admin should be able to reveal credentials in catalog-based MCP servers. I find it kind of spooky tbh. I would be more in favor of just duplicating basically all of these handlers so that we can have a better separation of concerns, but there would certainly be a lot of duplicated code. I am open to doing this tomorrow if we think it's a better idea than what I have here.